driver-core/src/main/com/mongodb/operation/ChangeStreamOperation.java

```diff
@@ -205,38 +256,159 @@ public Collation getCollation() {
+                                                getCommand(connection.getDescription(), binding.getSessionContext()),
+                                                CommandResultDocumentCodec.create(RAW_BSON_DOCUMENT_CODEC, FIELD_NAMES_WITH_RESULT),
+                                                connection, asyncTransformer(source, connection),
+                                                new SingleResultCallback<AsyncBatchCursor<RawBsonDocument>>() {
+                                                    @Override
+                                                    public void onResult(final AsyncBatchCursor<RawBsonDocument> result,
+                                                                         final Throwable t) {
+                                                        if (t != null) {
+                                                            wrappedCallback.onResult(null, t);
+                                                        } else {
+                                                            wrappedCallback.onResult(
+                                                                    new AsyncChangeStreamBatchCursor<T>(ChangeStreamOperation.this,
+                                                                            result, binding), null);
+                                                        }
+                                                    }
+                                                });
+                                    }
+                                }
+                            });
                 }
             }
         });
     }
 
-    private AggregateOperation<RawBsonDocument> createAggregateOperation() {
+    private BsonDocument getCommand(final ConnectionDescription description, final SessionContext sessionContext) {
+        BsonValue aggregate = changeStreamLevel == ChangeStreamLevel.COLLECTION
+                ? new BsonString(namespace.getCollectionName()) : new BsonInt32(1);
+        BsonDocument commandDocument = new BsonDocument("aggregate", aggregate);
+        appendReadConcernToCommand(sessionContext, commandDocument);
+
         List<BsonDocument> changeStreamPipeline = new ArrayList<BsonDocument>();
         BsonDocument changeStream = new BsonDocument("fullDocument", new BsonString(fullDocument.getValue()));
+
+        if (changeStreamLevel == ChangeStreamLevel.CLIENT) {
+            changeStream.append("allChangesForCluster", BsonBoolean.TRUE);
+        }
+
         if (resumeToken != null) {
             changeStream.append("resumeAfter", resumeToken);
         }
+
+        BsonTimestamp startAtTime = startAtOperationTime;
+        if (startAtTime == null && resumeToken == null && serverIsAtLeastVersionFourDotZero(description)) {
+            if (sessionContext.getOperationTime() != null
+                    && sessionContext.getOperationTime().compareTo(description.getOperationTime()) > 0) {
```

Comment:
I can't think of a case where the description's operationTime is null and server version is >= 4.0, so I think this is ok.